gradle-baseline-java/src/test/java/com/palantir/baseline/BaselineTestingIntegrationTest.java

```diff
@@ -0,0 +1,327 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline;
+
+import static com.palantir.gradle.testing.assertion.GradlePluginTestAssertions.assertThat;
+
+import com.palantir.gradle.testing.execution.GradleInvoker;
+import com.palantir.gradle.testing.execution.InvocationResult;
+import com.palantir.gradle.testing.junit.GradlePluginTests;
+import com.palantir.gradle.testing.project.RootProject;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import org.junit.jupiter.api.Test;
+
+@GradlePluginTests
+public class BaselineTestingIntegrationTest {
+
+    private void standardBuildFile(RootProject rootProject) {
```

Comment:
I think this would be better as a @BeforeEach method. Should we maybe add this to the guide?